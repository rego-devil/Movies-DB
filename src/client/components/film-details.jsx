import React from 'react';
import { Link } from 'react-router-dom';
import Loading  from './loading';

export class FilmDetails extends React.Component {

  componentWillMount() {
    this.props.getFilm(this.props.match.params.id);
  }

  render() {
    const {currentFilmDetails, onClearCurrentFilm, isFetching} = this.props;    

    if(!currentFilmDetails) 
      return null

    return (
      <div className="filmDetails">
       
        <img className="filmDetails__img"  src={currentFilmDetails.poster_path} />
        <div className="filmDetails__overview">
          <div className="filmDetails__top">
            <div className="filmDetails__title">{currentFilmDetails.title}</div>
            <div className="filmDetails__rating">{currentFilmDetails.vote_average}</div>
            
              <Link to="/">
                <button className="filmDetails__backToSearch" onClick={() => onClearCurrentFilm(null)}>
                  Back to search
                  </button>
              </Link>
            
          </div>
          <div className="filmDetails__middle">
            <div className="filmDetails__date">{currentFilmDetails.release_date && currentFilmDetails.release_date.split('-')[0]}</div>
            <div className="filmDetails__duration">{currentFilmDetails.runtime ? currentFilmDetails.runtime + ` min.` : ''}</div>
          </div>
          <div className="filmDetails__description">{currentFilmDetails.overview}</div>
          <div className="filmDetails__genres">
          {
            currentFilmDetails.genres && currentFilmDetails.genres.map((item, i) => (i > 0 ? ', ':'') + item)
          }
          </div>
        </div>
      </div>
      
      
      // currentFilmDetails ? (
      //     result
      // ) : (isFetching ? 
      //   <React.Fragment>
      //     { console.log('----3----')}
      //     <Loading />
      //   </React.Fragment> : '')
        
    )
  }
  
}