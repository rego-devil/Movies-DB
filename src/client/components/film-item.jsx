import React from 'react';
import propTypes from 'prop-types';


export const FilmItem = ({title, genres, release_date, poster_path}) => {

	let genreList = genres.map((item,i) => {
		return (i > 0 ? ', ':'') + item;
	});

	return (
		<React.Fragment>
			<img src={poster_path} className="filmItem__img" />
			<div className="filmItem__description">
					<div className="filmItem__title">{title}</div>
					<div className="filmItem__releaseYear">{release_date.split('-')[0]}</div>
			</div>
			<div className="filmItem__genre">{genreList}</div>
		</React.Fragment>
	)
}

FilmItem.propTypes = {
	title: propTypes.string.isRequired,
	release_date: propTypes.string.isRequired,
	poster_path: propTypes.string.isRequired,
	genres: propTypes.array,
}