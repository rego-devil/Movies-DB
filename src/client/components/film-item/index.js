export { default } from './film-item';
