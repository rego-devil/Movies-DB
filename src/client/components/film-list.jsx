import React from 'react';
import {FilmItem, FilmSort} from './';

export class FilmList extends React.Component {
	constructor() {
		super();
	}

	render() {
		const {filmList, onShowFilmDetails, onSelectFilm} = this.props;		
		return (
			filmList.length > 0 ? (
				<React.Fragment>
				<FilmSort />
				<ul className="filmList">
					{
						filmList.map((item) => 
							<li key={item.id} className="filmItem" onClick={() =>  onShowFilmDetails(item) }>
								<FilmItem {...item} />
							</li>
						)
					}
				</ul>
				</React.Fragment>
			) : (
				<div className="noFilms">No films found</div>
			)
		)
	}
}


