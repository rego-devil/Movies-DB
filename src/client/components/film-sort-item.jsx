import React from 'react';


export const SortItem = ({sortBy, onSort, currentSortField, value, sortOrderLocal }) => (
	<span 
		className={`filmSort__item ${currentSortField === sortBy ? 'filmSort__item--active' : ''}`} 
		onClick={() => onSort({sortBy: currentSortField, sortOrder: sortOrderLocal}) } 
	>
		{value}
		{currentSortField === sortBy ? sortOrderLocal === 'asc' ? String.fromCharCode(8593) : String.fromCharCode(8595) : ''  }
	</span>
)