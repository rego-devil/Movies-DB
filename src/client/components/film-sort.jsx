import React from 'react';

export const FilmSort = () => (
  <div className="filmInfo">
		<div className="filmInfo__count">7 film found</div>
		<div className="filmInfo__sortBy">Sort by: 
			<span className="filmInfo__sortByActive">release date</span>
			<span>rating</span>
		</div>
	</div>
)