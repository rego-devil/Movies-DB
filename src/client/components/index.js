export { Header } from './header';
export { FilmItem } from './film-item';
export { FilmList } from './film-list';
export { FilmDetails } from './film-details';
export { FilmSort } from './film-sort';
export { SortItem } from './film-sort-item';
export { Search } from './search';
export { Footer } from './footer';
