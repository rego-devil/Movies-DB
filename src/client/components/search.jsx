import React from 'react';
import {SearchFilterItem} from './';
import {Context} from './context';

export class Search extends React.Component {
	constructor(props) {
		super(props);

		this.handleInput = this.handleInput.bind(this);
		this.handleActive = this.handleActive.bind(this);
		

		this.state = {
			inputValue: '',
			searchBy: 'Title',
			selectedTab: 0
		}
		
	}

	handleInput(e) {
		this.setState({inputValue: e.target.value})
	}

	handleActive(value, number) {
		this.setState({
			searchBy: value,
			selectedTab: number
		});
	}


	render() {
		const {inputValue, classActive, selectedTab} = this.state;
		const {onSearch} = this.props;
		return (
			<div className="search">
				<div className="search__title">Find your Movie</div>
				<input type="text" className="search__input" value={inputValue} onInput={this.handleInput} />
				<div className="search__bottom">
					<div className="search__filter">
						<span className="search__filterText">Search By:</span>
						{ 
							['title', 'director'].map((item, i) => 
								<SearchFilterItem key={item} name={item} isActive={ i===selectedTab }  onActive={() => this.handleActive(item, i)} />
							)
						}
					</div>
					<button className="search__button" onClick={onSearch}>Search</button>
				</div>
			</div>
		)
	}
}
