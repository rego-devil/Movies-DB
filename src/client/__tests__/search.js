import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Search } from '../components/search';

describe('Search component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });

  test('setState ', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper.state('search')).toEqual('');
    expect(wrapper.find('.search__input').prop('value')).toEqual('');

    wrapper.setState({ search: 'Some text in input' });

    expect(wrapper.find('.search__input').prop('value')).toEqual('Some text in input');
  });

  test('input data in textBox', () => {
    const wrapper = mount(<Search />);
    wrapper.find('.search__input').simulate('change', {
      target: { value: 'Input some text' },
    });

    expect(wrapper.state('search')).toBe('Input some text');
  });
});
