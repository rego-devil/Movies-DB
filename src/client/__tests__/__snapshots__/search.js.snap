// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders without crashing 1`] = `
<div
  className="search"
>
  <div
    className="search__title"
  >
    Find your Movie
  </div>
  <input
    className="search__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="search__bottom"
  >
    <div
      className="search__filter"
    >
      <span
        className="search__filterText"
      >
        Search By:
      </span>
      <SearchFilterItem
        isActive={true}
        key="title"
        name="title"
        onActive={[Function]}
      />
      <SearchFilterItem
        isActive={false}
        key="director"
        name="director"
        onActive={[Function]}
      />
    </div>
    <button
      className="search__button"
    >
      Search
    </button>
  </div>
</div>
`;
